libr/bin/format/dex/dex.c

```diff
@@ -56,10 +78,26 @@ struct r_bin_dex_obj_t* r_bin_dex_new_buf(RBuffer *buf) {
 	}
 	bin->header.class_size = classes_size / sizeof (struct dex_class_t);
 	bin->classes = (struct dex_class_t *) malloc (classes_size);
-	r_buf_read_at (bin->b, bin->header.class_offset, (ut8*)bin->classes, classes_size);
+	for (i = 0; i < bin->header.class_size; i++) {
+		bin->classes[i].class_id = r_read_le32 (bin->b->buf + bin->header.class_offset +
+						i * sizeof (struct dex_class_t) + 0);
+		bin->classes[i].access_flags = r_read_le32 (bin->b->buf + bin->header.class_offset +
```

Comment:
what about use r_offsetof. and maybe deifne a macro to make all this stuff much more readable?